kernel/arch/dreamcast/sound/snd_stream.c

```diff
@@ -151,16 +151,40 @@ static void process_filters(snd_stream_hnd_t hnd, void **buffer, int *samplecnt)
 }
 
 static void sep_data(void *buffer, int len, int stereo) {
-    uint32_t data;
     uint32_t *buf = (uint32_t *)buffer;
-    uint16_t *left = (uint16_t *)sep_buffer[1];
-    uint16_t *right = (uint16_t *)sep_buffer[0];
+    uint32_t *left_ptr = (uint32_t *)sep_buffer[1];
+    uint32_t *right_ptr = (uint32_t *)sep_buffer[0];
+    uint32_t data;
+    uint32_t left_val;
+    uint32_t right_val;
 
     if(stereo) {
-        for(; len > 0; len -= 2) {
+        len <<= 1;
+        for(; len > 8; len -= 8) {
+            if(((uintptr_t)buf & 31) == 0) {
```

Comment:
I believe you can remove the `if`. The `pref` instruction will ignore the lowest 5 bits of the address.

Reply:
Yeah I can remove this if statement, thanks.